flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/source/StreamingMonitorFunction.java

```diff
@@ -155,13 +178,21 @@ private void monitorAndForwardSplits() {
       if (lastSnapshotId == INIT_LAST_SNAPSHOT_ID) {
         newScanContext = scanContext.copyWithSnapshotId(snapshotId);
       } else {
+        snapshotId = maxReachableSnapshotId(lastSnapshotId, snapshotId,
+            scanContext.maxSnapshotCountPerMonitorInterval());
         newScanContext = scanContext.copyWithAppendsBetween(lastSnapshotId, snapshotId);
       }
 
+      LOG.debug("Start discovering splits from {}(exclusive) to {}(inclusive),", lastSnapshotId, snapshotId);
```

Comment:
This is missing spaces between the snapshot IDs and the inclusive or exclusive clarification.